File: linked-list-merge/LinkedListMerge.java
import java.util.*;

class LinkedListMerge {
  public static class Node {
    int value;
    Node next;

    Node(int value) {
      this.value = value;
      this.next = null;
    }
  }

  public static Node mergeLinkedLists(Node headOne, Node headTwo) {
    Node head = new Node(0);
    Node current = head;

    while(headOne != null && headTwo != null) {
        if(headOne.value < headTwo.value) {
            current.next = headOne;
            headOne = headOne.next;
            current = current.next;
        } else {
            current.next = headTwo;
            headTwo = headTwo.next;
            current = current.next;
        }
    }
    if(headOne != null) {
        current.next = headOne;
    }
    if(headTwo != null) {
        current.next = headTwo;
    }
    return head.next;
  }
}

File: linked-list-merge/LinkedListMerge.swift
class LinkedListMerge {
    class Node {
        var next: Node?
        var value: Int
        init(value: Int){ self.value = value }
    }

    func linkedListMerge(_ headOne: Node, _ headTwo: Node) -> Node {
        var current: Node = LinkedList(value: 0)
        var result: Node? = current
        var head1: Node? = headOne
        var head2: Node? = headTwo
        
        while head1 != nil, head2 != nil {
            if head2!.value < head1!.value {
                result!.next = head2
                head2 = head2!.next
                result = result!.next
            } else {
                result!.next = head1
                head1 = head1!.next
                result = result!.next
            }
        }
        if(head1 != nil) { result!.next = head1 }
        if(head2 != nil) { result!.next = head2 }
        return current.next!
    }
}